src/views/Home/Assignment/index.tsx

```diff
@@ -0,0 +1,223 @@
+import Icon from '#rscg/Icon';
+import Pager from '#rscv/Pager';
+import Button from '#rsca/Button';
+
+import useRequest from '#utils/request';
+import { notifyOnFailure } from '#utils/requestNotify';
+
+import {
+    ProjectElement,
+    MultiResponse,
+} from '#typings';
+
+import _ts from '#ts';
+
+import styles from './styles.scss';
+
+interface Assignment {
+    id: number;
+    createdAt: string;
+    projectDetails: ProjectElement;
+    createdByDetails: {
+        id: number;
+        displayName: string;
+        email: string;
+    };
+    contentObjectDetails: {
+        id: number;
+        title: string;
+    };
+    isDone: boolean;
+    contentObjectType: {
+        id: number;
+        title: string;
+    };
+}
+
+interface Data {
+    key: number;
+    assignee: string;
+    sourceTitle: string;
+    sourceURL: string;
+    projectName: string;
+    date: string;
+}
+
+interface Props {
+}
+
+interface AssignmentRendererProps extends Assignment {
+    handleClick: (id: number) => void;
+}
+
+const emptyLink = '#';
+const maxItemsPerPage = 5;
+
+function assignmentRenderer(info: AssignmentRendererProps) {
+    const handleClick = () => {
+        info.handleClick(info.id);
+    };
+    return (
+        <div className={styles.assignmentItem}>
+            <a
+                className={styles.link}
+                href={emptyLink}
+            >
+                {info.createdByDetails.displayName}
+            </a>
+            <span> {_ts('assignment', 'assignedYou')} </span>
+            <a
+                className={styles.link}
+                href={emptyLink}
+            >
+                {info.contentObjectDetails?.title}
+            </a>
+            <span> {_ts('assignment', 'in')} </span>
+            <a
+                href={emptyLink}
+                className={styles.link}
+            >
+                {info.projectDetails.title}
+            </a>
+            <div className={styles.inline}>
+                <FormattedDate
+                    className={styles.date}
+                    value={info.createdAt}
+                    mode="hh:mm aaa, MMM dd, yyyy"
+                />
+                <Button
+                    transparent
+                    iconName="checkCircle"
+                    className={styles.icon}
+                    onClick={handleClick}
+                />
+            </div>
+        </div>
+    );
+}
+
+const keySelector = (info: Assignment) => info.id;
+
+
+function Assignments(props: Props) {
+    const [activePage, setActivePage] = useState<number>(1);
+    const [assignmentCount, setAssignmentCount] = useState<number>(0);
+    const [selectedAssignment, setSelectedAssignment] = useState<number>();
+    const [
+        pending,
+        assignmentsResponse,
+        ,
+        getAssignments,
+    ] = useRequest<MultiResponse<Assignment>>(
+        {
+            url: 'server://assignments/',
+            method: 'GET',
+            query: {
+                is_done: 3,
```

Comment:
What does `3` mean?

Reply:
Unfortunately, instead of 'true' or 'false', is_done is assigned numbers (1: unknown, 2: true, 3: false) from the server itself.

Reply:
Could you add a comment in the code on what the numbers mean